lib/ansible/cli/galaxy.py

```diff
@@ -1420,17 +1420,12 @@ def execute_login(self):
             if C.GALAXY_TOKEN:
                 github_token = C.GALAXY_TOKEN
             else:
-                login = GalaxyLogin(self.galaxy)
-                github_token = login.create_github_token()
+                github_token = GitHubOAuthTokenCreator().github_interactive_device_flow_auth()
         else:
             github_token = context.CLIARGS['token']
 
         galaxy_response = self.api.authenticate(github_token)
 
-        if context.CLIARGS['token'] is None and C.GALAXY_TOKEN is None:
-            # Remove the token we created
-            login.remove_github_token()
-
```

Comment:
If this needs to be done in two months we may want to keep this emitting a warnings until then, so that the bugfix versions of Ansible would warn users at least for some time...

Reply:
I think we are actually leaning toward not "fixing" this, and just having `ansible-galaxy login` error, and give instructions for manually getting the token from the galaxy UI.